torch/csrc/autograd/engine.cpp

```diff
@@ -228,14 +236,16 @@ static variable_list call_post_hooks(Function& fn, variable_list outputs, variab
 }
 
 static variable_list call_function(FunctionTask& task) {
+  bool prev_checkpoint_valid_state = checkpoint_valid;
+  checkpoint_valid = task.base->can_checkpoint() && prev_checkpoint_valid_state;
```

Comment:
It would still be nice to use a RAII style struct to avoid messing up code that shouldn't need to worry about this (3rd time)